lib/internal/Magento/Framework/Controller/Result/Json.php

```diff
@@ -59,10 +60,20 @@ public function setJsonData($jsonData)
     }
 
     /**
-     * {@inheritdoc}
+     * @param HttpResponse|ResponseInterface $response
+     * @return $this
      */
     protected function render(ResponseInterface $response)
     {
+        return $this->renderHttpResponse($response);
+    }
+
+    /**
+     * @param HttpResponse $response
+     * @return $this
+     */
+    private function renderHttpResponse(HttpResponse $response)
```

Comment:
True. Since all classes extending `\Magento\Framework\Controller\AbstractResult`  are HTTP specific result instances that would work fine. Should I update the PR accordingly?